### DISCUSSION TITLE
The `hover:` class does not exist. If `hover:` is a custom class, make sure it is defined within a `@layer` directive

### DISCUSSION THREAD

**Discussion by DebabrataSaha-570** [tailwindlabs/tailwindcss#8754] [Help]
I am new to tailwind css. I got this error while building a react project. I found a similar problem on StackOverflow but It isn't helping me. Can anyone tell me why this is happening?

```
//Index.css 
@tailwind base;
@tailwind components;
@tailwind utilities;


@layer components {
    .btn-purple {
        @apply px-4 py-1 text-sm text-purple-600 font-semibold border border-purple-200 rounded-full hover: text-white hover:bg-purple-600 hover:border-transparent focus:outline-none focus:ring-2 focus:ring-purple-600 focus:ring-offset-2;
    }
}
```
I asked the same question on StackOverflow : 
https://stackoverflow.com/questions/72810279/the-hover-class-does-not-exist-if-hover-is-a-custom-class-make-sure-it-i/72810642#72810642

**Comment by adamwathan** [comment]
You have a space between `hover:` and `text-white`.

**Reply by themodernmonk7** [reply]
The space is added automatically after saving.

**Comment by HorizonChaser12** [comment]
I am also getting the same issue.The Tailwind intellisense formats the code and adds a space even after i remove the space.If you got any solution please post it up here.Thank you!!

**Reply by akkhan00** [reply]
go to settings in vscode and search for 'format on save' and then remove the check marks and it will work.

**Reply by TanjimulSabbir** [reply]
it's working. Thanks

**Comment by luisfridev** [comment]
I had the same problem using visual studio code, i "solved it" changing the "language mode" to **tailwindcss** instead just **css**

![image](https://user-images.githubusercontent.com/2937237/189214049-8ad972e0-9134-4bc8-846d-ca92cb27189b.png)

**Reply by mybearworld** [reply]
This is the correct solution. You can also go to your settings.json and add this:

```json
"files.associations": {
  "*.css": "tailwindcss"
},
```

So this is done by default.

**Comment by themodernmonk7** [comment]
Hey, If you go the solution then share it here. 
I am still struggling.

**Comment by mathurinfarms** [comment]
Hey guys, it's possible that VSC automatically adds a space after "hover:" but there are some settings that you can change in order to avoid this behavior.

Personally, I leave this on and I save using CTRL+K, and then CTRL+SHIFT+S. This is the VSC shortcut to Save without formatting:
![image](https://user-images.githubusercontent.com/97749828/204103111-129f0694-cb06-49bb-b361-0cc5393e020e.png)

I hope this helps. Cheers!

**Reply by akhror-valiev** [reply]
thanks, it was the most straightforward approach and saved time

**Reply by mathurinfarms** [reply]
> thanks, it was the most straightforward approach and saved time

Glad it helped. It's a pleasure to help out. Cheers!

**Comment by jzzfs** [comment]
The same goes for breakpoint utils like `sm:` and `md:`. The whitespace gets added upon autosave and escaping the colon doesn't help either -> ...The `sm\:absolute` class does not exist...

**Comment by nomoreerrors** [comment]
I just run the "npm run build" command to build css-file and it works

**Comment by sarjid** [comment]
**If You Are Using Visual Studio Code, You Can Follow These Step**

Create A Directory in Your Application
Like this   **.vscode/settings.json** 

Now copy and paste this code in your **settings.json** file , I hope it will work
```
{
  "css.lint.unknownAtRules": "ignore",
  "[css]": { "editor.defaultFormatter": "bradlc.vscode-tailwindcss" }
}
```
![solve 1](https://github.com/tailwindlabs/tailwindcss/assets/53226726/875d5582-a4e4-4fdc-8777-cc1c9c03d91b)
![code solution 2](https://github.com/tailwindlabs/tailwindcss/assets/53226726/96634c2e-4143-4895-9790-618a4f1d5d88)

**Comment by ramkole** [comment]
<img width="442" alt="image" src="https://github.com/tailwindlabs/tailwindcss/assets/56620810/f27bedad-553b-4765-8a99-7184988e3378">

removing space worked for me
